enterprise/api/v1/views/enterprise_user.py

```diff
@@ -0,0 +1,66 @@
+"""
+Views for the ``enterprise-user`` API endpoint.
+"""
+
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework import permissions, response, status
+
+from django.core.exceptions import ValidationError
+
+from enterprise import models
+from enterprise.api.v1 import serializers
+from enterprise.api.v1.views.base_views import EnterpriseReadOnlyModelViewSet
+from enterprise.logging import getEnterpriseLogger
+
+LOGGER = getEnterpriseLogger(__name__)
+
+
+class EnterpriseUserViewSet(EnterpriseReadOnlyModelViewSet):
+    """
+    API views for the ``enterprise-user`` API endpoint.
+    """
+    queryset = models.EnterpriseCustomerUser.objects.all()
+    filter_backends = (DjangoFilterBackend,)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    USER_ID_FILTER = 'enterprise_customer_users__user_id'
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        - Filter down the queryset of groups available to the requesting uuid.
+        """
+        enterprise_uuid = kwargs.get('enterprise_uuid', None)
+
+        try:
+            enterprise_customer_queryset = models.EnterpriseCustomerUser.objects.filter(
+                enterprise_customer__uuid=enterprise_uuid
+            )
+
+            if enterprise_customer_queryset.exists():
+                serializer = serializers.EnterpriseUserSerializer(enterprise_customer_queryset, many=True)
+                return response.Response(serializer.data)
```

Comment:
(issue): we should return a list of both pending and existing enterprise users. I think how the code is structured currently, it's returning one or the other.

I think we can try something like

```
class EnterpriseUserViewSet(EnterpriseReadOnlyModelViewSet):
    """
    API views for the ``enterprise-user`` API endpoint.
    """
    queryset = models.EnterpriseCustomerUser.objects.all()
    filter_backends = (DjangoFilterBackend,)
    permission_classes = (permissions.IsAuthenticated,)

    USER_ID_FILTER = 'enterprise_customer_users__user_id'

    def retrieve(self, request, *args, **kwargs):
        """
        - Filter down the queryset of groups available to the requesting uuid.
        """
        enterprise_uuid = kwargs.get('enterprise_uuid', None)
        users = []
        try:
            enterprise_customer_queryset = models.EnterpriseCustomerUser.objects.filter(
                enterprise_customer__uuid=enterprise_uuid,
            )
            users.extend(enterprise_customer_queryset)
            pending_enterprise_customer_queryset = models.PendingEnterpriseCustomerUser.objects.filter(
                enterprise_customer__uuid=enterprise_uuid
            ).order_by('user_email')
            users.extend(pending_enterprise_customer_queryset)
        except ValidationError:
            # did not find UUID match in either EnterpriseCustomerUser or  PendingEnterpriseCustomerUser
            return response.Response(
                {'detail': 'Could not find enterprise uuid {}'.format(enterprise_uuid)},
                status=status.HTTP_404_NOT_FOUND
            )
        serializer = serializers.EnterpriseUserSerializer(
            users,
            many=True
        )
        serializer_data = sorted(
            serializer.data, key=lambda k: k['is_admin'], reverse=True)
        return response.Response(serializer_data)
```

Reply:
(issue): apologies for not mentioning this in the ticket but could we also support searching for an individual user name or email?

Reply:
> (issue): we should return a list of both pending and existing enterprise users. I think how the code is structured currently, it's returning one or the other.
> 

Apologies, but my understanding from our initial discussion was that a user could be an enterprise customer user or a pending customer user - but not both. Is that no longer the case? Both of these datasets come from two different models thus I had split them up accordingly

But happy to make the changes if necessary